### DISCUSSION TITLE
Prevent blocked users from commenting on commits

### DISCUSSION THREAD

**Discussion by ghost** [community/community#64948] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

When we block a user, they cannot write a comment under the GitHub issue we created, but the blocked user can write a comment under the commits. The blocked user should not be able to comment under the commits.
Some users continue to bully by commenting on commits because they were blocked. (When the issue we created is resolved by the owner of the repo, the issue is closed by specifying a commit. I `m talking about this.)

Could you please make a improving to also prevent them from posting comments under commits?

**Comment by TahirAhmad01** [comment]
As of my last knowledge update in September 2021, GitHub's blocking functionality primarily prevents the blocked user from interacting with your repositories directly, such as creating issues, pull requests, or commenting on them. However, it might not prevent them from commenting on commits made in public repositories.

If you're experiencing issues with users continuing to bully by commenting on commits even after being blocked, here are a few suggestions you could consider:

1. **Report the Behavior**: If a blocked user is engaging in harmful behavior like bullying through commit comments, you should report this behavior to GitHub's support team. They might be able to take appropriate action to address the issue.

2. **Moderation**: Depending on the severity of the situation, you could consider enabling moderation for your repository's discussions or commit comments. This way, all comments would need approval before they become visible to others. However, this could slow down legitimate contributions as well.

3. **Privatize the Repository**: If the project isn't open source and you're facing persistent issues from certain users, consider making the repository private. This way, only approved collaborators can access and interact with the repository's content.

4. **Use GitHub Actions**: You might be able to create a custom GitHub Action that automatically hides or removes comments made by blocked users on commits. This would require some programming and GitHub Actions knowledge. You could trigger this action on certain events like new commits being pushed.

5. **Feature Request**: Reach out to GitHub's support or community forums to suggest an enhancement to the blocking feature that also prevents blocked users from commenting on commits. They might consider implementing such functionality in the future if it's deemed beneficial.

6. **Documentation Check**: It's always good to check GitHub's official documentation or the platform's latest updates to see if any new features related to blocking and commenting have been introduced since my last update in September 2021.

Remember, it's crucial to maintain a healthy and respectful environment in open source communities. If you're facing persistent issues with harassment or bullying, reach out to GitHub's support and your community for assistance.
